### PR TITLE
python-i3ipc: init at 1.3.0

### PR DESCRIPTION
**Pull request by 1pakch**
###### Motivation for this change

The package [i3ipc-python](https://github.com/ziberna/i3-py) is a better-maintained alternative to [i3-py](https://github.com/ziberna/i3-py) which is currently in nixpkgs. Contributors: 26 vs 3.  Last commit: 19 days ago vs 6 years ago

###### Things done

- [ ] Tested using sandboxing
  - Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested the library
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/i3ipc/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ pkgs, buildPythonPackage, enum-compat }:
```
Comment (FRidh) [inline pkgs/development/python-modules/i3ipc/default.nix]: no `pkgs`

File: pkgs/development/python-modules/i3ipc/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+let
+  version = "1.3.0";
+  sha256 = "1rw9mq18np6bf8xp8vvc21bi5q8xjmj8dck2vsa1hy8bk434259m";
+
+in buildPythonPackage rec {
+  name = "i3ipc-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/i3ipc/default.nix]:
```
pname = "i3ipc";
version = "1.3.0";
```
no need for name

File: pkgs/development/python-modules/i3ipc/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = pkgs.fetchFromGitHub {
+    owner = "acrisci";
+    repo = "i3ipc-python";
+    rev = "refs/tags/v${version}";
+    inherit sha256;
+    fetchSubmodules = true; # because a tag is used
+  };
+
+  doCheck = false;
```
Comment (FRidh) [inline pkgs/development/python-modules/i3ipc/default.nix]: Why? Leave a comment in the code with a reason.

File: pkgs/development/python-modules/i3ipc/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  propagatedBuildInputs = [ enum-compat ];
+
+  meta = {
+    homepage = "https://github.com/acrisci/i3ipc-python";
+    description = "An improved library to control i3wm";
+    license = pkgs.lib.licenses.bsd3;
+  };
```
Comment (FRidh) [inline pkgs/development/python-modules/i3ipc/default.nix]: are you going to maintain this expression?

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/i3ipc/default.nix (hunk 4 above)
Comment (ilya-kolpakov) [reply pkgs/development/python-modules/i3ipc/default.nix]: Yes. Added myself as maintainer here and in lib/maintainers.nix

File: pkgs/development/python-modules/i3ipc/default.nix (hunk 1 above)
Comment (ilya-kolpakov) [reply pkgs/development/python-modules/i3ipc/default.nix]: fixed

File: pkgs/development/python-modules/i3ipc/default.nix (hunk 2 above)
Comment (ilya-kolpakov) [reply pkgs/development/python-modules/i3ipc/default.nix]: fixed

File: pkgs/development/python-modules/i3ipc/default.nix (hunk 3 above)
Comment (ilya-kolpakov) [reply pkgs/development/python-modules/i3ipc/default.nix]: Added a comment

File: pkgs/development/python-modules/i3ipc/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+in buildPythonPackage rec {
+  pname = "i3ipc";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "acrisci";
+    repo = "i3ipc-python";
+    rev = "refs/tags/v${version}";
+    inherit sha256;
+    fetchSubmodules = true; # Fetching by tag does not work otherwise
```
Comment (FRidh) [inline pkgs/development/python-modules/i3ipc/default.nix]: seems to fetch just fine here without. Note also that `refs/tags` can be dropped.

**Comment by samueldr** [comment]
(triage) This is now part of `nixpkgs`, see #34835.

* * * 
<sup>If for any reason this needs to be re-opened, re-open, or ask for it.</sup>

**Pull request closed without merging**